lib/improver/tests/test_ensemble_copula_coupling_GeneratePercentilesFromProbabilities.py

```diff
@@ -0,0 +1,502 @@
+                         [[[-20.8, -16., -11.2],
+                           [-6.4, -1.6, 3.2],
+                           [8., 8.33333333, 8.66666667]]],
+                         [[[9., 9.33333333, 9.66666667],
+                           [10., 10.5, 11.],
+                           [11.5, -37., -34.]]],
+                         [[[-31., -28., -25.],
+                           [-22., -19., -16.],
+                           [-13., -10., -7.]]],
+                         [[[-4., -1., 2.],
+                           [5., 8., 8.5],
+                           [9., 9.5, -37.6]]],
+                         [[[-35.2, -32.8, -30.4],
+                           [-28., -25.6, -23.2],
+                           [-20.8, -18.4, -16.]]],
+                         [[[-13.6, -11.2, -8.8],
+                           [-6.4, -4., -1.6],
+                           [0.8, 3.2, 5.6]]]])
+        cube = self.current_temperature_forecast_cube
+        percentiles = np.arange(0.05, 1.0, 0.05)
+        bounds_pairing = (-40, 50)
+        plugin = Plugin()
+        result = plugin._probabilities_to_percentiles(
+            cube, percentiles, bounds_pairing)
+        self.assertArrayAlmostEqual(result.data, data)
+
+    def test_check_data_spot_forecasts(self):
+        """
+        Test that the plugin returns an Iris.cube.Cube with the expected
+        data values for the percentiles for spot forecasts.
+        """
+        data = np.array([[[15.8, 31., 46.2,
+                           8., 10., 31.,
+                           10.4, 12., 42.4]],
+                         [[-16., 10, 31.,
+                           8., 10., 11.6,
+                           -30.4, 8., 12.]],
+                         [[-30.4, 8., 11.,
+                           -34., -10., 9,
+                           -35.2, -16., 3.2]]])
+        cube = self.current_temperature_spot_forecast_cube
+        percentiles = [0.1, 0.5, 0.9]
+        bounds_pairing = (-40, 50)
+        plugin = Plugin()
+        result = plugin._probabilities_to_percentiles(
+            cube, percentiles, bounds_pairing)
+        self.assertArrayAlmostEqual(result.data, data)
+
+
+class Test__convert_bounds_units(IrisTest):
+
+    """Test the _convert_bounds_units plugin."""
+
+    def setUp(self):
+        self.current_temperature_forecast_cube = (
+            _add_forecast_reference_time_and_forecast_period(
+                set_up_temperature_cube()))
+
+    def test_basic(self):
+        """Test that the result is a numpy array."""
+        cube = self.current_temperature_forecast_cube
+        plugin = Plugin()
+        result = plugin._convert_bounds_units(cube)
+        self.assertIsInstance(result, np.ndarray)
+
+    def test_check_data(self):
+        """
+        Test that the expected results are returned for the bounds_pairing.
+        """
+        cube = self.current_temperature_forecast_cube
+        fp_units = (
+            cube.coord("probability_above_threshold").units)
```

Comment:
Can't you just manually create the expected results? Same thing applies to the test below.